mods/patterns/init.lua

```diff
@@ -0,0 +1,34 @@
+patterns = {}
+
+patterns.check_limit = function(term, table)
+    local state = false
+    if (table == nil) then
+        state = true
+    else
+        for i=1, #table do
+            if (term == table[i]) then
+                state = true
+            end
+        end
+    end
+    return state
+end
```

Comment:
There's a bit of extra things here, and this check doesn't actually work as you intend.

both a `false` and `true` value here will return true with `if patterns.check_limit(colors[1], limit) then`
^ the check there checks for nil values (not true vs false), and since your function doesn't return a nil value, it always bypasses this check.

You don't need a `state` value, straightaway `return true`
as for a false value, you don't need to provide that as well, it will automatically returns nil and that is classified as a false factor under this check; `if patterns.check_limit(colors[1], limit) then`

Reply:
EDIT: I missed that you bracketed the check, that works.
But the previous point still stands, the false value is redundant. You can just work with a true/nil value here.